src/ONYXKEYS.js

```diff
@@ -87,6 +87,9 @@ export default {
     // Indicates which locale should be used
     PREFERRED_LOCALE: 'preferredLocale',
 
+    // List of transactions that are in progress of being rejected
+    REJECT_IN_PROGRESS: 'rejectInProgress',
```

Comment:
I think we can come up with a slightly better KEY name. Some things we should think about are:
1. It should be plural since this is an array (list) of ids, not just one value
2. Since it stores transaction ids, maybe that should be part of the name. REJECT_IN_PROGRESS kinda feels like it should be a boolean to me, if I didn't already know the purpose of the data stored here
3. In the future there will be many, many things that can be rejected. To keep this future-proof, maybe we can get the word "IOU" in the key name?

Some ideas of names:
- IOU_TRANSACTION_IDS_BEING_REJECTED (this is probably too long)
- IOU_TRANSACTIONS_BEING_REJECTED / *_CANCELLED

What do you think?

cc @Julesssss

Reply:
I agree.
IOU_TRANSACTIONS_BEING_REJECTED sounds good to me.

Reply:
I agree and these suggestions seem fine. 

Perhaps there isn't a need for a totally new key here, as rejecting transactions are (currently) limited to IOU transactions. Any reason for not including this value within the [existing IOU key](https://github.com/Expensify/Expensify.cash/pull/3498/files/2aa493b144cf298c2dc63d45aae0aeef41a38a2b#diff-daa865a27f5c734becece9770029064b7cd62b56034d54cb2eb0279fa6cc548cR31)?

Reply:
umm I don't get you. We're just renaming the key REJECT_IN_PROGRESS to  IOU_TRANSACTIONS_BEING_REJECTED.

> I agree and these suggestions seem fine.
> 
> Perhaps there isn't a need for a totally new key here, as rejecting transactions are (currently) limited to IOU transactions. Any reason for not including this value within the [existing IOU key](https://github.com/Expensify/Expensify.cash/pull/3498/files/2aa493b144cf298c2dc63d45aae0aeef41a38a2b#diff-daa865a27f5c734becece9770029064b7cd62b56034d54cb2eb0279fa6cc548cR31)?

Reply:
I'm suggesting that you don't need a new key here. We hold multiple values within a single key, as [shown here](https://github.com/Expensify/Expensify.cash/blob/1951973dca9a374fbacb53c88a5dbf24edfbefe0/src/libs/actions/IOU.js#L64).

Reply:
Makes sense. Is this what you mean?
IOU: {
    + **_transactionsBeingRejected: Array of strings._**
    creatingIOUTransaction: Boolean
    error: Boolean
    loading: Boolean
}

Reply:
Yeah, exactly. Any thoughts on which solution would be preferred? Personally i think it makes sense to collect all the IOU 'loading' states under one Onyx key.

Reply:
Yeah definitely, all under one key is more clean.
The only reason for new key might be so that all kinds of transaction rejection can use it.

Reply:
Cool. So we'll have multiple types of report rejection in the future, but there's no need to optimize for that now.

Reply:
Alright I tried my best, I think Onyx doesn't support what I'm trying to do.
So we should go ahead with creating a new key IOU_TRANSACTIONS_BEING_REJECTED.

Reply:
Yeah sadly you're right, at the moment there's no "set one property in a key without updating the others" unless you pass the rest of the object with it.

You could connect `ReportTransaction` to `ONYXKEYS.IOU` and pass the entire iou object to `rejectTransaction`, update the property `transactionsBeingRejected` before and after the API call, and set the entire old object back in the key, maybe - I think that's pretty similar to how you have it now, except reusing the `iou` key

Reply:
Okay, how do I connect `ReportTransaction` to `ONYXKEYS.IOU` and get the object?
My futile attempt below :(
```
// ReportTransaction.js inside the class.
let iou;
Onyx.connect({
    key: ONYXKEYS.IOU,
    callback: val => iou = val,
});
```

Reply:
Aah so here's what I was thinking:

In `ReportTransaction.js`...
1. Change `key: ONYXKEYS.REJECT_IN_PROGRESS,` to `key: ONYXKEYS.IOU` & change `rejectInProgress` to `iou` (since that's what we're doing elsewhere in the app)
2. pass `iou: this.props.iou` in `rejectTransaction`
3. In `isRejected` just get `this.props.iou.transactionsBeingRejected` and do your checks on that array

In `IOU.js`...
1. Accept the new param (`iou` instead of `rejectedIDs`)
2. In `.finally()` do something like:
```js
const withoutCurrentID = _.get(iou, 'transactionsBeingRejected', []).filter(id => id !== transactionID);
Onyx.set(ONYXKEYS.IOU, {...iou, transactionsBeingRejected: withoutCurrentID});
```

What do you think about that?

Reply:
Looks great! Finally things are working as they should :)
Btw, underscore's `get` was only introduced in 1.12.0 and we use 1.10.2.
We could update, but I'll use lodashGet for now.

Reply:
Ooh good call about `get`, I didn't know that! Hopefully we update some day soon ish haha